src/main/java/com/rapid7/client/dcerpc/mslsad/messages/LsarQueryInformationPolicyResponse.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2017, Rapid7, Inc.
+ *
+ * License: BSD-3-clause
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * * Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ */
+package com.rapid7.client.dcerpc.mslsad.messages;
+
+import java.io.IOException;
+import com.rapid7.client.dcerpc.io.PacketInput;
+import com.rapid7.client.dcerpc.io.ndr.Unmarshallable;
+import com.rapid7.client.dcerpc.messages.RequestResponse;
+import com.rapid7.client.dcerpc.mslsad.objects.POLICY_INFORMATION_CLASS;
+
+public class LsarQueryInformationPolicyResponse<T extends Unmarshallable> extends RequestResponse {
```

Comment:
We can bound it by POLICY_INFORMATION instead of Unmarshallable if we use the shared parent class/interface

Reply:
It really doesn't buy you much, since as a user of this response you need to call out the exact object you want. With unions, there tends not to be any common information which would cater itself to use inheritance on the Java side. Would only serve to be self documenting here.